pkg/maps/metricsmap/metricsmap.go

```diff
@@ -183,23 +177,88 @@ func SyncMetricsMap() error {
 
 	var key, nextKey Key
 	for {
-		err := metricsmap.GetNextKey(&key, &nextKey)
+		err := bpf.GetNextKey(metricsmap.GetFd(), unsafe.Pointer(&key), unsafe.Pointer(&nextKey))
 		if err != nil {
 			break
 		}
-		entry, err := metricsmap.Lookup(&nextKey)
+		err = bpf.LookupElement(metricsmap.GetFd(), unsafe.Pointer(&nextKey), unsafe.Pointer(&entry))
 		if err != nil {
 			return fmt.Errorf("unable to lookup metrics map: %s", err)
 		}
-		value := entry.(*Value)
-		// Increment Prometheus metrics here.
-		updatePrometheusMetrics(&nextKey, value)
-		key = nextKey
+
+		// cannot use `range entry` since, if the first value for a particular
+		// CPU is zero, it never iterates over the next non-zero value.
+		for i := 0; i < possibleCpus; i++ {
+			// Increment Prometheus metrics here.
+			updatePrometheusMetrics(&nextKey, &entry[i])
+			key = nextKey
```

Comment:
Probably this can be done just once, rather than once per cpu.

Reply:
As commented above, this does not send any network traffic and simply updates the local cached value, so will be keeping this logic.

Reply:
My bad.. Yes I will change that.

Reply:
done